test/go/ferry_test.go

```diff
@@ -37,6 +32,31 @@ func (t *FerryTestSuite) TestReadOnlyDatabaseFailsInitialization() {
 	ferry = testhelpers.NewTestFerry().Ferry
 	err = ferry.Initialize()
 	t.Require().Nil(err)
+
+	_, err = t.Ferry.TargetDB.Exec("SET GLOBAL read_only = OFF")
+	t.Require().Nil(err)
```

Comment:
Why is this added?

Reply:
TL;DR I've removed the teardown, and inlined that code into the specific test that required it.

These lines were initially part of the test suite teardown. Previously, the test suite contained a single test, so it didn't matter whether this code lived in the teardown, or in the test itself.

Now, I've added a second test which fails because of the teardown. 

Ghostferry requires that the target DB is not a replica (read only OFF). The original test set the target DB to read only, performed assertions, and then turned read only off.

Without this change, the first test may run, followed by the newly-added second test. The teardown is run at the end of the suite, so the read only flag is still on when the second test runs, and thus fails.

We should make sure we set the target DB back to its correct state, before moving on to the next test.